app/controllers/answers_controller.rb

```diff
@@ -1,11 +1,37 @@
 class AnswersController < ApplicationController
 
-  def index
+  def new
+    @question = Question.find(params[:question_id])
+    @answer = Answer.new
   end
 
-  def show
+  def create
+    question = Question.find(params[:question_id])
+    answer = question.answers.new(content: params[:answer][:content], user_id: session[:user_id])
+    if answer.save
+      redirect_to question
+    else
+      # Add ajax to display the error message getting back from validations
+      redirect_to new_question_answer_path(question.id)
+    end
   end
 
-  def create
+  def edit
+    @answer = Answer.find(params[:id])
+  end
+
+  def update
+    answer = Answer.find(params[:id])
+    question = answer.question
+    if answer.update(content: params[:answer][:content])
+      redirect_to question
+    else
+      redirect_to :back
```

Comment:
I don't know whether this is ajax or not, but I think the right behavior here is render "edit" ?